app/presenters/tree_builder_ops_features.rb

```diff
@@ -0,0 +1,249 @@
+class TreeBuilderOpsFeatures < TreeBuilder
+  include CompressedIds
+
+  has_kids_for Hash, [:x_get_tree_hash_kids]
+
+  attr_reader :role, :features, :editable
+
+  def initialize(name, type, sandbox, build, role:, features:, editable: false)
+    @role     = role
+    @editable = editable
+    @features = features
+
+    super(name, type, sandbox, build)
+  end
+
+  private
+
+  def feature_tree
+    recurse_menu_items(filtered_menus, root_select)
+  end
+
+  def filtered_menus
+    menus = []
+
+    Menu::Manager.each do |section|
+      next if section.id == :cons && !Settings.product.consumption
+      next if section.name.nil?
+      next unless Vmdb::PermissionStores.instance.can?(section.id)
+
+      menus.push(section)
+    end
+
+    menus
+  end
+
+  def recurse_menu_items(items, checked = false)
+    items.map do |item|
+      case item
+      when Menu::Section
+        menu_section_to_node(item)
+      when Menu::Item
+        if item.feature.nil? || !MiqProductFeature.feature_exists?(item.feature)
+          next
+        end
+        menu_item_to_node(item, checked)
+      when String
+        feature_name_to_node(item, checked)
+      else
+        item
+      end
+    end
+  end
+
+  def menu_section_to_node(section)
+    node = default_node.merge!(
+      :id      => "#{node_id}___tab_#{section.id}",
+      :text    => _(section.name),
+      :tooltip => _("%{title} Main Tab") % {:title => section.name},
+      :data    => {}
+    )
+
+    node[:data][:kids] = recurse_menu_items(section.items).compact
```

Comment:
I'd really like to do it the proper way but this tree builder framework is not very clear. Like, why are `_tree_kids` methods plural if it's only supposed to return one kid? What do you mean by the "constructed from the top"? 

A HUGE part of the complexity here is determining if a node should be checked. (It doesn't even work correctly right now.) It's unclear how to tie parents and children together _and_ count checkboxes. Basically the tree has to be assembled before each node is converted. 

I spent longer than I felt was appropriate trying to squeeze this menu tree into `x_get_tree_roots` and `x_get_tree_hash_kids`, finally giving up and doing things more like they currently are (recursively building the menu). 

I do feel like this code is a bit kludge to work around x_get_* methods. With TreeBuilder it's like this thing has to be built twice. What is the better solution?